samples/unit-tests/maps/series-map/demo.js

```diff
@@ -8,24 +8,61 @@ QUnit.test('Mapping of joinBy with custom sub options', function (assert) {
         series: [
             {},
             {
-                data: [
-                    {
-                        custom: {
-                            code: 'au-nt'
-                        },
-                        value: 0
-                    }
-                ],
-                mapData: Highcharts.maps['countries/au/au-all'],
-                joinBy: ['hc-key', 'custom.code']
+                data: [{
+                    custom: {
+                        code: 'PL' // iso-a2 (by code, inside custom)
+                    },
+                    value: 0
+                }, {
+                    value: 0,
+                    code: 'no' // hc-key (by code)
+                }, {
+                    value: 0,
+                    'hc-key': 'dk' // hc-key (directly by string)
+                }, {
+                    value: 0,
+                    'iso-a2': 'SE' // iso-a2 (directly by string)
+                }],
+                mapData: Highcharts.maps['custom/europe'],
+                joinBy: ['iso-a2', 'custom.code']
             }
         ]
+    }),
+    series = chart.series[1];
+
+    assert.strictEqual(
+        series.points[0].graphic.attr('fill'),
+        'rgb(128,142,166)',
+        'joinBy with custom sub options - territory should be colored.'
+    );
+
+    [
+        ['hc-key', 'code'],
+        'hc-key',
+        'iso-a2',
+        ['iso-a2', 'custom.code']
+    ].forEach((joinBy, i) => {
+        assert.strictEqual(
+            series.points[i].graphic.attr('fill'),
+            'rgb(128,142,166)',
+            'Different joinBy updates - territory should be colored.'
+        );
+
+        series.update({ // #15374
+            joinBy: joinBy
+        });
+
+        assert.strictEqual(
+            series.points[i].color,
+            Highcharts.seriesTypes.map.defaultOptions.nullColor,
+            'The joinBy changed - previous territory should not be colored.'
+        );
```

Comment:
Add something to the asserts messages based on the `forEach` loop so in case of a failure it will be clear what exactly failed. Even using `i` should be enough.